pages/favoriteList.js

```diff
@@ -0,0 +1,59 @@
+import styled from "styled-components";
+import Head from "next/head";
+import OutivitiesListItem from "@/components/OutivitiesListItem";
+
+export default function FavoritesList({
+  outivities,
+  onToggleFavorite,
+  favorites,
+}) {
+  const favoriteOutivities = outivities.filter((outivity) =>
+    favorites.includes(outivity.id)
+  );
+
+  return (
+    <>
+      <Head>
+        <title>My Outivities</title>
+      </Head>
+
+      <StyledTitle>Favorite Outivities</StyledTitle>
+
+      <main>
+        {favoriteOutivities.length === 0 && (
+          <>
+            <p>You have not saved any favorite Outivity yet.</p>
+            <p>Start adding your first favorite!</p>
+          </>
+        )}
+
+        <StyledOutivityCardContainer>
```

Comment:
You should also render this ``<ul>`` conditionally. Because now it is rendered even if there is no favorite. You can render either the message above or the List.

Reply:
Very good point! We changed this.